pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -454,38 +531,86 @@ async def run(self):
+                    record_checksum_type = getattr(CHECKSUM_TYPES, record.checksum_type.upper())
+                    checksum_types[record.type] = record_checksum_type
+                    record.checksum_type = record_checksum_type
+
+                    if not self.mirror and record.type not in types_to_download:
+                        continue
 
-                    if record.type in types_to_download:
-                        sanitized = urlpath_sanitize(self.remote_url, record.location_href)
-                        downloader = self.remote.get_downloader(url=sanitized)
-                        metadata_downloaders[record.type] = asyncio.ensure_future(downloader.run())
+                    downloader = self.remote.get_downloader(
+                        url=urlpath_sanitize(self.remote_url, record.location_href),
+                        expected_size=record.size,
+                        expected_digests={record_checksum_type: record.checksum},
+                    )
+                    repomd_downloaders[record.type] = asyncio.ensure_future(
+                        run_repomdrecord_download(record.type, record.location_href, downloader)
+                    )
 
                 self.repository.original_checksum_types = checksum_types
 
                 try:
-                    for future in asyncio.as_completed(list(metadata_downloaders.values())):
-                        await future
+                    for future in asyncio.as_completed(list(repomd_downloaders.values())):
+                        name, location_href, result = await future
+                        store_metadata_for_mirroring(self.repository, result, location_href)
+                        repomd_files[name] = result
                         metadata_pb.increment()
                 except ClientResponseError as exc:
                     raise HTTPNotFound(reason=_("File not found: {}".format(exc.request_info.url)))
 
-                metadata_files = {key: val.result() for key, val in metadata_downloaders.items()}
-                await self.parse_repository_metadata(repomd, metadata_files)
+                try:
+                    downloader = self.remote.get_downloader(
+                        url=urlpath_sanitize(self.remote_url, "repodata/repomd.xml.asc")
+                    )
+                    result = await downloader.run()
+                    store_metadata_for_mirroring(self.repository, result, "repodata/repomd.xml.asc")
+                    metadata_pb.increment()
+                except ClientResponseError:
+                    pass
+
+                try:
+                    downloader = self.remote.get_downloader(
+                        url=urlpath_sanitize(self.remote_url, "extra_files.json")
+                    )
+                    result = await downloader.run()
+                    store_metadata_for_mirroring(self.repository, result, "extra_files.json")
```

Comment:
I think we do need to (or should) download extra_files for license compliance reasons.  CentOS repo metadata is itself GPL licensed, and we are redistributing it.  Granted this only helps if the license file is declared properly in extra_files.json